### ISSUE TITLE
[Issue] .NET SDK Container publishing should use /t:PublishContainer instead of /p:PublishProfile=DefaultContainer

### ISSUE THREAD

**Issue by baronfel** [Azure/azure-dev#3426]
**Describe the bug**

The [`azd` source](https://github.com/Azure/azure-dev/blob/5dce2f60b4956ec31f9268ecd0dded1324fafdad/cli/azd/pkg/tools/dotnet/dotnet.go#L155-L179) is containerizing  .NET applications via a command like:

```
dotnet publish <project> -r linux-x64 -c Release -p PublishProfile=DefaultContainer -p ContainerImageName=some_name -p ContainerRegistry=some_registry
```

This will work for Web and SDK projects, but it will not work for Console applications. What _will_ work for every kind of containerizable project is using `/t:PublishContainer` instead of `-p PublishProfile=DefaultContainer`. I won't go into too much detail, but the mechanism that 'translates' `-p PublishProfile=DefaultContainer` into `/t:PublishContainer` doesn't exist and will likely never exist for the vast majority of console projects, and it's simpler and clearer to just invoke the correct MSBuild target anyway.

**Comment by ellismg** [comment]
Just confirming my understanding from some of the linked bugs, @baronfel: If customers run into issues with this target not existing, they just need to upgrade their `dotnet` SDK to 8.0.200 or later?

**Comment by baronfel** [comment]
Yeah, that's right - does `azd` have knowledge of what SDK the user is using? That could be used to safely pivot the behavior.
